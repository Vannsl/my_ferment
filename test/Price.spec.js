import { shallowMount } from '@vue/test-utils'
import Price from '@/components/Price'

const factory = () => {
  return shallowMount(Price, {
    propsData: {
      name: 'Pro',
      list: ['Thing', 'Thing', 'Thing', 'Thing'],
      price: '29,99 €',
      limited: false
    }
  })
}

describe('Prices', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has the property name', () => {
    const wrapper = factory()
    expect(wrapper.props('name')).toBe('Pro')
  })

  it('has the property list', () => {
    const wrapper = factory()
    expect(wrapper.props('list')).toEqual(['Thing', 'Thing', 'Thing', 'Thing'])
  })

  it('has the property price', () => {
    const wrapper = factory()
    expect(wrapper.props('price')).toBe('29,99 €')
  })

  it('has the property limited', () => {
    const wrapper = factory()
    expect(wrapper.props('limited')).toBe(false)
  })
})
