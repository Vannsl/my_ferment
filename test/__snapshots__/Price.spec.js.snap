// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prices renders properly 1`] = `
"<div class=\\"flex flex-col w-5/6 lg:w-1/4 mx-auto lg:mx-0 rounded-none lg:rounded-l-lg bg-white mt-4\\"><div class=\\"flex-1 bg-white text-gray-600 rounded-t rounded-b-none overflow-hidden shadow\\"><div class=\\"p-8 text-3xl font-bold text-center border-b-4\\">Pro</div> <ul class=\\"w-full text-center text-sm\\"><li class=\\"border-b py-4\\">Thing</li><li class=\\"border-b py-4\\">Thing</li><li class=\\"border-b py-4\\">Thing</li><li class=\\"border-b py-4\\">Thing</li></ul></div> <div class=\\"flex-none mt-auto bg-white rounded-b rounded-t-none overflow-hidden shadow p-6\\"><div class=\\"w-full pt-6 text-3xl text-gray-600 font-bold text-center\\">
      29,99 â‚¬
      <span class=\\"text-base\\">/ per user</span></div> <div class=\\"flex items-center justify-center\\"><button class=\\"mx-auto lg:mx-0 hover:underline gradient text-white font-bold rounded-full my-6 py-4 px-8 shadow-lg\\">Sign Up</button></div></div></div>"
`;
